src/main/scala/org/constellation/primitives/CheckpointBlock.scala

```diff
@@ -79,10 +79,20 @@ case class CheckpointBlock(
     this.copy(checkpoint = checkpoint.copy(edge = checkpoint.edge.withSignature(hs)))
 
   def plus(other: CheckpointBlock): CheckpointBlock =
-    this.copy(checkpoint = checkpoint.plus(other.checkpoint))
+    this.copy(
+      checkpoint = checkpoint.plus(other.checkpoint),
+      transactions = transactions ++ other.transactions,
+      messages = messages ++ other.messages,
+      notifications = notifications ++ other.notifications,
+      observations = observations ++ other.observations
+    )
+
+  def +(other: CheckpointBlock): CheckpointBlock = plus(other)
 
-  def +(other: CheckpointBlock): CheckpointBlock =
-    this.copy(checkpoint = checkpoint.plus(other.checkpoint))
+  def plusEdge(other: CheckpointBlock): CheckpointBlock =
```

Comment:
I do not know why but I do not like something in that method :D Probably here is the reason:

`+`/`plus` for CheckpointBlock takes CheckpointBlock and returns CheckpointBlock (I am wondering if it should't be a Semigroup btw).

But here we have method which just adds edge to checkpoint but takes `CheckpointBlock` so we have kinda conflict between method's name and parameter's name. So I would either:

```scala
def plusEdge(edge: CheckpointEdge): CheckpointBlock = this.copy(checkpoint = checkpoint + other.checkpoint)
```

or 

```scala
def plusEdgeFrom(other: CheckpointBlock): CheckpointBlock = this.copy(checkpoint = checkpoint + other.checkpoint)
```